packages/esm-appointments-app/src/home-appointments/appointments-list.component.tsx

```diff
@@ -184,12 +212,35 @@ const AppointmentsBaseTable = () => {
       content: (
         <span className={styles.serviceContainer}>
           <RenderStatus status={appointment.status} appointmentUuid={appointment.id} t={t} mutate={mutate} />
-          <ActionsMenu appointment={appointment} useBahmniUI={useBahmniUI} />
         </span>
       ),
     },
   }));
 
+  const handleFilter = ({ rowIds, headers, cellsById, inputValue, getCellId }: FilterProps): Array<string> => {
+    return rowIds.filter((rowId) =>
+      headers.some(({ key }) => {
+        const cellId = getCellId(rowId, key);
+        const filterableValue = cellsById[cellId].value;
+        const filterTerm = inputValue.toLowerCase();
+
+        if (typeof filterableValue === 'boolean') {
+          return false;
+        }
+        if (filterableValue.hasOwnProperty('content')) {
+          if (Array.isArray(filterableValue.content.props.children)) {
+            return ('' + filterableValue.content.props.children[1].props.children).toLowerCase().includes(filterTerm);
+          }
+          if (typeof filterableValue.content.props.children === 'object') {
+            return ('' + filterableValue.content.props.children.props.children).toLowerCase().includes(filterTerm);
```

Comment:
In think `props.children` is dangerous, since this could be a JSX element. This could cause a white screen if the UI changes.